portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java

```diff
@@ -1910,6 +1910,7 @@ public void includeLogin(PortalRenderContext rcontext, HttpServletRequest req,
 					message = rloader.getFormattedMessage("sit_return", impersonatorDisplayId);
 					rcontext.put("impersonatorDisplayId", impersonatorDisplayId);
 				}
+				rcontext.put("userDisplayName", Validator.escapeHtml(UserDirectoryService.getCurrentUser().getDisplayName())); 
```

Comment:
Why do we need this?

Reply:
I need it to know what the actual user name is, because the announcements don't save the id or the eid to get in the lesson > announcement content, this use only the displayname (the actual user name), and in the javascripts of announcement content in lesson i couldn't get the displayname from another way